### DISCUSSION TITLE
Disable the frustrating "Press Ctrl + I to ask Copliot to do something" message on new files and EVERY new cell in a notebook

### DISCUSSION THREAD

**Discussion by bfgdigital** [community/community#67656] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

The recent change that copilot now pops up everywhere is driving me insane.
I know it's there, I know how to use it, I don't want to see it, I only want to see code suggestions.

Every single time a new cell is created in a python notebook, that damn message pops up again.

It's becoming too in-my-face and I now, for the first time, find myself turning Copilot off because it's annoying.

The setting options are to only turn off the useful features not the annoying, invasive features like these unwelcome messages - so instead the whole thing is canned and disabled and I switch to CodeWhisperer.

Please keep it invisible.
![image](https://github.com/community/community/assets/47546089/20b0e354-d931-48e7-90ce-9522d2f9e97c)

**Comment by xxFREESHROUDxx** [comment]
Hello @bfgdigital ,

here's what you can try to potentially disable those messages:

Check Copilot Settings: First, review Copilot's settings within your code editor. It's possible that there might be an option to disable or customize these messages.

Editor Settings: Sometimes, these messages are controlled by your code editor itself. Depending on the editor you're using, you might find an option to disable these pop-ups in the editor's settings.

Extensions/Plugins: If you're using an editor like Visual Studio Code, check if there are any Copilot-related extensions or plugins installed. Sometimes, these can provide additional settings or options to control Copilot behavior.

Submit Feedback: If you can't find a direct way to disable these messages, consider submitting feedback to GitHub or Copilot developers. They might not be aware of the extent of the annoyance this is causing and could potentially address it in future updates.

**Reply by bfgdigital** [reply]
This is literally where the copilot extension sends you to submit feedback (and this is feedback).
I tried those options (VS Code) they don't seem responsible for the annoying messages.
Cheers.

**Reply by ArbibSoufiane** [reply]
Answer from [ziishaned](https://github.com/ziishaned)
I found a way to hide the text that is displayed when the file is empty. To disable the empty state message, open your VS Code settings.json file and add the following content inside it.
"workbench.editor.empty.hint": "hidden"

**Reply by sat-ch** [reply]
"workbench.editor.empty.hint": "hidden" 
this works. Thank you!

**Reply by cztomsik** [reply]
Also set "inlineChat.lineNaturalLanguageHint": false, either this one does the trick, or you now need both.

**Reply by valentin-huebner** [reply]
@cztomsik I needed `"inlineChat.lineNaturalLanguageHint": false` but did not need `"workbench.editor.empty.hint": "hidden"`

**Reply by Xiol** [reply]
Big fan of the first response to this issue being clearly LLM generated and offering absolutely no value to the conversation at all.

**Reply by yurkovych** [reply]
> "inlineChat.lineNaturalLanguageHint": false

THANK YOU!

**Reply by Sudheer104** [reply]
> Answer from [ziishaned](https://github.com/ziishaned) I found a way to hide the text that is displayed when the file is empty. To disable the empty state message, open your VS Code settings.json file and add the following content inside it. "workbench.editor.empty.hint": "hidden"

Thanks this truly work "workbench.editor.empty.hint": "hidden"

**Comment by spitis** [comment]
Please, has anyone found a solution for this? I want to use copilot, but this is annoying af.

**Reply by rushi055** [reply]
workbench.editor.empty.hint: "hidden"

try this in settings.json file 

"github.copilot.chat.useProjectTemplates": false,
    "github.copilot.chat.welcomeMessage": "never",
    "workbench.editor.empty.hint": "hidden"

**Reply by kasir-barati** [reply]
It was so nerve wracking for me that I decided to get rid of it all together. My next stop is cody

**Comment by FahriBilici** [comment]
I have the same problem. it's annoying everytime I clicked the cell, it automatically opens the chat

**Comment by stekatag** [comment]
I have the same problem as well.

**Comment by Roon627** [comment]
Anybody got an answer ?

**Comment by illusionistlynx** [comment]
This is also driving me crazy. I have toggled every setting, disabled every possible interaction and language, etc. Nothing seems to turn this off. On top of that, there appears to be absolutely no way to disable Copilot Chat at all, even though it's a BETA release. Unacceptable. If you are part of an Organization or Enterprise you can disable it for everyone, but not as an individual.

**Comment by thangckt** [comment]
vote to disable this feature

**Comment by vasiljevic** [comment]
There is the option github.copilot.chat.welcomeMessage ("Control when to show the detailed welcome chat message") and we just need similar one for the "ask Copilot" message.

**Comment by ulayera** [comment]
Just cancelled my subscription and uninstalled the extension because of this. Irresponsible decision to override vscode's default message and not adding an easy to find way to disable it.

Edit: For more context, disabling the message doesn't help, because it will also disable the original VSCode Hint that allows you to choose the language of the new file you're creating.

**Comment by ziishaned** [comment]
I found a way to hide the text that is displayed when the file is empty. To disable the empty state message, open your VS Code `settings.json` file and add the following content inside it.

```json
"workbench.editor.empty.hint": "hidden"
```

**Reply by rgs2151** [reply]
Thank you!!

**Reply by sanzgadea** [reply]
Thanks, that worked for me as well

**Reply by khanhmay004** [reply]
thank so much !!!

**Reply by scottsb** [reply]
There's also an option in the Preferences UI for this. Just search "empty hint" to pull it up quickly.

**Comment by dawid-niedzwiecki** [comment]
PLEASE DISABLE IT FOR GODS SAKE IT'S SUCH AN EASY THING TO DO

**Reply by dawid-niedzwiecki** [reply]
Or just change the shortcut.............

**Comment by alexdev888** [comment]
same problem in vs code terminal

**Reply by wheelerjl** [reply]
> same problem in vs code terminal

I've got it turned off with this setting. Right clicking the text gave me a "Do you want to disable this?" option that added this setting to my settings.json

`"terminal.integrated.initialHint": false,`

**Comment by chana987** [comment]
For anyone who got here because of inline hints being preluded with `Press Ctrl+I to continue with Github Copilot...` in VSCode, disable in settings:
```
Github › Copilot › Chat › Inline Chat Hint: Enabled
   Hint for inline chat that shows once a line mostly consists of words.
```

**Reply by VinhNgT** [reply]
> For anyone who got here because of inline hints being preluded with `Press Ctrl+I to continue with Github Copilot...` in VSCode, disable in settings:
> 
> ```
> Github › Copilot › Chat › Inline Chat Hint: Enabled
>    Hint for inline chat that shows once a line mostly consists of words.
> ```

Seems like it changed
![image](https://github.com/user-attachments/assets/28513ba8-cc94-49c8-b3dc-1b727cbcb257)

**Reply by rafjohnson** [reply]
Thanks. Looks like the "Press Ctrl+I..." message ignores the "Disable Completions" setting for the window. This did remove the message.

**Comment by sueszli** [comment]
It's literally just `"inlineChat.lineNaturalLanguageHint": false,.`.

All suggestions above are wrong.

**Reply by cztomsik** [reply]
Yes, but `"workbench.editor.empty.hint": "hidden"` worked before.

**Reply by athena-adam** [reply]
Thank you!!  the "workbench.editor.empty.hint": "hidden" seemed to work for a lot of people, but not me.
Your solution, `"inlineChat.lineNaturalLanguageHint": false` works perfectly!  I can now begin to rebuild my sanity.

**Reply by pierreBodineau** [reply]
Why is an experimental feature activated by default? Beats me...

**Comment by spitis** [comment]
Just use Cursor guys. It's way better than copilot.

**Comment by mahnunchik** [comment]
How to completely disable this annoying message?

**Comment by mubaidr** [comment]
It can be disabled by following setting:

```jsond
"inlineChat.lineEmptyHint": false
```

**Comment by euanvandenberg** [comment]
Remove it from empty files with:

"workbench.editor.empty.hint": "hidden"

And from the terminal with:

"terminal.integrated.initialHint": false

**Comment by slspencer** [comment]
Simple UI solutions (maybe not technically simple):
1. 'x' at the right end of the Copilot prompt closes the prompt
2. define "Ctrl-I" to toggle the Copilot prompt on/off

**Comment by JiteshJangra** [comment]
right click on that message it gives an opton to diable it

**Reply by PMaackFT** [reply]
But now it no longer gives me the option to select the language I want to use. Which is the thing I was trying to get back. Do you have any idea how to make that come back?

**Comment by verbess** [comment]
Right-click that message to disable inline chat hint, or add the following to your `settings` file:  

``` json
"inlineChat.lineNaturalLanguageHint": false,
```

**Comment by cygnuson** [comment]
It would be better to not enable AI slop without asking, but these comments help remove this particular piece.

**Reply by king-ayanda** [reply]
Agreed. Or at least add one button that completely removes all AI slop. Too bad.

**Comment by safakamali** [comment]
I and many other people want this text not to be displayed and instead to show an option for selecting the language used like before and it show this window when clicking on it:
![image](https://github.com/user-attachments/assets/f08756ce-ea8f-4bb7-9738-396875248d26)

**Reply by PMaackFT** [reply]
Agreed. I really would like the option to use the AI features but having them break the fundamental functionality of being able to pick the language is a deal-breaker for me. It boggles the mind that someone thought it was a good idea to take a thing that existed and replace it with a completely different thing. Why would they not just tack the new thing on the end or above/below it? It's like someone putting new tires on their car, only to find out the new tires make it impossible to open the door, so now you gotta roll the windows down and climb in like the Dukes of Hazard.

**Reply by PMaackFT** [reply]
Ok, so... A coworker of mine just told me about a handy little feature I'd literally never noticed before. You can select the language from the bottom right (Next to the bell icon) and get the same dialog. I felt INCREDIBLY dumb for not having realized that before, and for my purposes that's an elegant solution to the "problem" I had with selecting a language on new document creation. Maybe it'll be helpful for some others as well.

**Comment by inovafani** [comment]
You guys want easy fix? check this out, trust me it works!

https://youtu.be/NxxUOaSmZ1Q

**Comment by khoalu** [comment]
`"workbench.editor.empty.hint": "hidden"` works, thanks

**Reply by lloydjatkinson** [reply]
This fixes it for me, I just wish that people didn't have to scroll so far down to see it.

Edit: I spoke too soon it came back again even with this. I'm just going to uninstall CoShit again.

**Comment by chriskathumbi** [comment]
I had to do so many checks to disable this stuff.
Please put all these chat functions as optional. 
I feel like I might just downgrade to the previous version.

**Comment by yassghn** [comment]
microsofts stole copilot. they stole the project, then microsoft legalized themselves theft of github code. one of the most massive theft projects in all of history. everyone who works on these generative content (it's not ai, they really wish it was ai, but it's not) projects needs to be removed from earth. we're slowly sunsetting msft corp, along with all of their assets, with a lot of arrests soon. bye bye, nazis!

**Reply by yassghn** [reply]
needless to say, but no one will pick up where msft left off. it's just gonna die. all of it. everything they stole, worked on, what have you.

**Comment by gelembungsahabat** [comment]
AI RUINING EVERYTHING

**Comment by bfgdigital** [comment]
Ok, I think we're done here. Thanks to everyone for your input.

**Comment by allefeld** [comment]
Turning off `inlineChat.lineNaturalLanguageHint` worked for me.

**Comment by archtaurus** [comment]
I close as much as possible about the AI. But:
Ctrl+N and it reads : "Use Ctrl +I to generate code" (now means May 2026)
Seems no setting can get rid of it? HELP ME, pls!
<img width="1208" height="184" alt="图片" src="https://github.com/user-attachments/assets/b0c0430f-2d48-4990-aa01-558f01d0281c" />
